sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py

```diff
@@ -222,7 +228,22 @@ def mlflow_tracking_uri(self) -> Optional[str]:
         :return: Returns mlflow tracking uri of the workspace.
         :rtype: str
         """
-        return self._mlflow_tracking_uri
+        # if _with_auth:
+        #     module_logger.warning(
+        #     "'_with_auth' is deprecated and will be removed in a future release. ")
+
+        try:
+            from azureml.mlflow import get_mlflow_tracking_uri_v2
+
+            return get_mlflow_tracking_uri_v2(self)
+        except ImportError as e:
+            error_msg = (
+                "azureml.mlflow could not be imported. "
+                "Please ensure that 'azureml-mlflow' has been installed in the current python environment."
+            )
+            raise UserErrorException(error_msg) from e
```

Comment:
We should not raise an error message if azureml-mlflow is not installed. It should be a warning message. Return the tracking uri. In warning say azureml-mlflow will not use credentials passed to MLCLient